src/zcl_abapgit_repo_srv.clas.abap

```diff
@@ -299,6 +319,45 @@ CLASS ZCL_ABAPGIT_REPO_SRV IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD zif_abapgit_repo_listener~on_meta_change.
+
+    DATA li_persistence TYPE REF TO zif_abapgit_persist_repo.
+
+    li_persistence = zcl_abapgit_persist_factory=>get_repo( ).
+    li_persistence->update_metadata(
+      iv_key         = iv_key
+      is_meta        = is_meta
+      is_change_mask = is_change_mask ).
+
+
+    " Recreate repo instance if type changed
+    IF is_change_mask-offline = abap_true.
```

Comment:
Previously it was needed due to the pull that happened automatically when creating a online repo, this does not happen anymore, so the code is not modified when removing and recreating online/offline repos

Reply:
Mmm, I wouldn't remove this feature. I personally use it. No really specific scenario, but sometime I add repo as offline and then attach it later after more parts are ready, Sometime, in opposite, detach  repo to avoid occasional issues. I think it is useful. Probably someone else also use it. We can move it deeper in UI but then the feature imho should stay.
Regarding the code: yeah, I'll add the comment ;) 
Or I propose to implement my suggestion from #1757 in the long run :)

Reply:
The user can do the same in 2 steps: Advanced -> Removed + Add new online/offline

But yea, we can move it to UI, have it call the SRV 2 times, then the method can be deleted from SRV, which will make it all look nicer, as then the LISTNER is not needed anymore

Reply:
Added more comments.

P.S. if you want to remove `switch` feature please check with others. They might be using the feature. I do definitely. Doing it via DB tool or repo recreation would be inconvenient. I'll think how to put it further in UI, maybe the main menu is not the right place indeed, but the feature itself is useful. It is like adding/removing origins in git - online/offline/attach/detach family is the direct analog in AG.

Reply:
suggest moving it one step up(to UI), so that

```abap
METHOD ui_switch_repo.

srv->delete_repo( ).

srv->create_repo_as_the_other_type()

ENDMETHDO.
```

this way both the SRV and REPO will be a lot cleaner

Reply:
UI: yeah, but I guess it is easier to solve than having the double indirection and dependency between srv and repo

Reply:
okay